From 85faaa5377e8aa4c8ece10fda9aa31f27dbd71a6 Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Fri, 27 Apr 2018 20:51:42 +0200
Subject: [PATCH 15/16] SystemUI: Toggle USB tethering only when USB is
 connected

Pressing the tile when unplugged causes USB tethering to turn on
as soon as the cable is connected. This can happen long after the
tile has been pressed, so don't do anything if unplugged.

Change-Id: Ib8bbdc758097ca7af5d4e7c3483c28ba42fb9ce1
---
 .../src/com/android/systemui/qs/tiles/UsbTetherTile.java      | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/UsbTetherTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/UsbTetherTile.java
index 84774b146f4..2a08142d8ea 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/UsbTetherTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/UsbTetherTile.java
@@ -75,7 +75,9 @@ public class UsbTetherTile extends QSTileImpl<BooleanState> {
 
     @Override
     protected void handleClick() {
-        mConnectivityManager.setUsbTethering(!mUsbTetherEnabled);
+        if (mUsbConnected) {
+            mConnectivityManager.setUsbTethering(!mUsbTetherEnabled);
+        }
     }
 
     @Override
-- 
2.17.1

